orders.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Orders endpoint
+ *
+ * LICENSE: The buyer can free use/edit/modify this software in anyway
+ * The buyer is NOT allowed to redistribute this module in anyway or resell it
+ * or redistribute it to third party
+ *
+ * @author    ROI Hunter Easy
+ * @copyright 2019 ROI Hunter
+ * @license   EULA
+ * @version   1.0
+ * @link      https://easy.roihunter.com/
+ */
+
+include(dirname(__FILE__) . '/../../config/config.inc.php');
+include(dirname(__FILE__) . '/../../init.php');
+require_once(_PS_MODULE_DIR_ . 'roihunter/classes/auth/authentication.php');
+require_once(_PS_MODULE_DIR_ . 'roihunter/roihunter.php');
+require_once(_PS_MODULE_DIR_. 'roihunter/classes/requests/OrdersResponse.php');
+
+ROIHunterAuthenticator::getInstance()->authenticate();
+
+const DEFAULT_LIMIT = 100;
+const DEFAULT_PAGE = 1;
+if ($_SERVER['REQUEST_METHOD'] == 'GET') {
+    try {
+        $dbOrdersResult = getCompletedOrdersForTheLastMonth(getPageParameter());
+    } catch (Exception $e) {
+        header("HTTP/1.1 500 Internal Server Error");
+        echo("Unknown server exception: " . $e->getMessage());
+        die();
+    }
+
+    header("HTTP/1.1 200 OK");
+    header("Content-Type:application/json");
+    echo(new OrdersResponse($dbOrdersResult));
+    die();
+} else {
+    header('HTTP/1.0 405 Method Not Allowed', true, 405);
+    die();
+}
+
+function getPageParameter() {
+    $page = Tools::getValue('page');
+    if (is_null($page) || empty($page)) {
+        return DEFAULT_PAGE;
+    } else {
+        if (!is_numeric($page)) {
+            header("HTTP/1.1 400 Bad Request");
+            echo "Page parameter is not valid.";
+            die();
+        } else if ((int) $page < 1) {
+            header("HTTP/1.1 400 Bad Request");
+            echo "Page parameter cannot be less than 1.";
+            die();
+        } else {
+            return (int) $page;
+        }
+    }
+}
+
+/**
+ * @throws PrestaShopDatabaseException
+ */
+function getCompletedOrdersForTheLastMonth($page) {
```

Comment:
We agreed with @martinfryc1 it would be nice to allow at least simply filtering by order status and creation date because  if somebody decide that we want orders from last 60 days not 30 it will impossible for us to get those data from existing customers.